<a href="https://colab.research.google.com/github/PinzonHarker/2024-II_InstrumentacionNuclear/blob/main/2_Taller_Radiactividad.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Previo

In [1]:
import numpy as np
import matplotlib as mpl
import matplotlib.pyplot as plt
import random as rn


def plot_template(title, x_label, y_label, x_lim=None, y_lim=None):
    """Genera la plantilla para mantener consistencia en el formato
    de cada tabla generada.

    Args:
        title (string): Aplica titulos.
        x_label (float): Título de eje x.
        y_label (float): Título de eje y.
        x_lim ([float], optional): Límite en eje x. Defaults to None.
        y_lim ([float], optional): Límite en eje y. Defaults to None.

    Returns:
        object: Devuelve los objetos de la gráfica modificados con la
        plantilla requerida
    """
    fig, ax = plt.subplots(  # Tamaño de la figura en pulgadas (ancho, alto)
        figsize=(10, 6)
    )

    # Configuración de rótulos y unidades
    ax.set_title(title, fontsize=16)  # Título con tamaño de fuente
    ax.set_xlabel(x_label, fontsize=14)  # Eje x con rótulo y unidades
    ax.set_ylabel(y_label, fontsize=14)  # Eje y con rótulo y unidades

    # Configuración de la escala y las dimensiones
    if x_lim is not None:
        ax.set_xlim(x_lim)  # Limitar el eje x
    if y_lim is not None:
        ax.set_ylim(y_lim)  # Limitar el eje y

    ax.grid(True, which="both", linestyle="--", alpha=0.7)  # Añadir una cuadrícula

    fig.tight_layout()  # Ajusta los elementos para que no se solapen

    return (  # Devolver las figuras y ejes para más modificaciones si es necesario
        fig,
        ax,
    )


# test
if False:
    x = np.linspace(0, 10, 100)
    fig, ax = plot_template("This is a test", "X label", "Y label", [0, 10], [-1, 1])
    ax.scatter(x, [np.sin(i) for i in x], marker="+", color="red")

# Simulación 2. Instrumentación Nuclear

Este trabajo resuelve la simulación #2 para la clase de Instrumentación Nuclear del semestre 2024-II a cargo del profesor L.F. Cristancho.

## Parte I: Distribución Exponencial.

A partir de la probabilidad

### 1. Núcleo elegido $^{60} \text{Co}$

In [ ]:
T12 = 5.2711   # [a]
u_T12 = 8e-4

# calculo de tau
tau = T12 / np.log(2)
u_tau = u_T12 / np.log(2)

# calculo de gamma
gamma = 1/tau
u_gamma = u_tau / tau**2

# imprimir
print(f"""
tau = ({tau:.4f} +- {u_tau:.4f}) [a]
gamma = ({gamma:.6f} +- {u_gamma:.6f}) [a^-1]
""")

# en segundos
tau *= 365 * 24 * 60 * 60
u_tau *= 365 * 24 * 60 * 60

# recalculo de gamma
gamma = 1/tau
u_gamma = u_tau / tau**2

# imprimir
print(f"""
tau = ({tau:.8e} +- {u_tau:.8e}) [s]
gamma = ({gamma:.8e} +- {u_gamma:.8e}) [s^-1]
""")


tau = (7.6046 +- 0.0012) [a]
gamma = (0.131500 +- 0.000020) [a^-1]


tau = (2.39818345e+08 +- 3.63974646e+04) [s]
gamma = (4.16982279e-09 +- 6.32858081e-13) [s^-1]



Se tomará el núcleo de Cobalto 60 debido al uso común en aplicaciones de radio terapia en la física médica. Las propiedades estudiadas se exponen a continuación : (14.6: Vida Media Y Cinética de Desintegración Radiactiva, 2022)

- Tiempo semivida $T_{1/2}$ = $5.2711 (8) \, \text{a}$

El cálculo de la constante de semidesintegración $\Gamma$ y la vida media $\tau$ se consiguen como (incluir referencia guía 1 profesor):

$$T_{1/2} = \ln (2) \, \tau \hspace{50pt} \Gamma = \frac{1}{\tau}$$

Luego, los parámetros resultan:
$$ \tau = 7.6046(12)  \, \text{a}  \hspace{50pt} \Gamma = 0.131500(20)  \, \text{a}^{-1}$$
$$ \tau = (2.39818(36)\times 10^8)  \, \text{s}  \hspace{50pt} \Gamma = (4.16982(63) \times 10^{-9})  \, \text{s}^{-1}$$

### 2. Periodo $\Delta t$

Elegimos el periodo de tiempo tal que nos de una cantidad razonable del promedio de decaimientos. Suponiendo una distribución Poisson del tipo

$$P_p(x) = \frac{\mu^x e^{-\mu}}{x}$$

Podemos suponer que la probabilidad individual por unidad de tiempo es de la forma
$$\Gamma = \frac{p}{\Delta t}$$

Para la probabilidad promedio que sería

$$\mu = Np = N \Gamma \Delta t$$



## Parte II: Distribución Binomial Núcleo por Núcleo

Se responderán a los resultados de la simulación de la desintegración nuclear como distribución binomial

### Probabilidad

## Referencias

- Indice de todo lo nuclear super bueno
Nuclear Data – Table – Laboratoire National Henri Becquerel. (2016). Lnhb.fr. http://www.lnhb.fr/home/nuclear-data/nuclear-data-table/

- Exacto para el 60
http://www.lnhb.fr/nuclides/Co-60_tables.pdf


- ToxFAQsTM: Cobalto (Cobalt) | ToxFAQ | ATSDR. (2023, February 2). Cdc.gov. https://www.atsdr.cdc.gov/es/toxfaqs/es_tfacts33.html#:~:text=El%2060Co%20(l%C3%A9ase%20cobalto,la%20investigaci%C3%B3n%20m%C3%A9dica%20y%20cient%C3%ADfica.

- 14.6: Vida media y cinética de desintegración radiactiva. (2022, October 30). LibreTexts Español. https://espanol.libretexts.org/Quimica/Qu%C3%ADmica_General/Mapa:_Qu%C3%ADmica_General:_Principios,_Patrones_y_Aplicaciones_%28Averill%29/14:_Cin%C3%A9tica_qu%C3%ADmica/14.06:_Vida_media_y_cin%C3%A9tica_de_desintegraci%C3%B3n_radiactiva?form=MG0AV3
